chord_metadata_service/discovery/fields.py

```diff
@@ -338,38 +330,55 @@ def filter_queryset_field_value(qs: QuerySet, field_props, value: str):
     the `mapping` value is based on the same model as the queryset.
     """
 
-    model, field = get_model_and_field(
+    model, field = f_utils.get_model_and_field(
         field_props["mapping_for_search_filter"] if "mapping_for_search_filter" in field_props
         else field_props["mapping"]
     )
 
+    group_by = field_props.get("group_by")
+
     if field_props["datatype"] == "string":
-        condition = {f"{field}__iexact": value}
+        if group_by:
+            # JSONField array string check must use 'contains' lookup
+            nested_condition = f_utils.get_nested_json_condition(group_by, value)
+            condition = Q(**{f"{field}__contains": [nested_condition]})
+        else:
+            condition = Q(**{f"{field}__iexact": value})
     elif field_props["datatype"] == "number":
         # values are of the form "[50, 150)", "< 50" or "≥ 800"
 
         if value.startswith("["):
             [start, end] = [int(v) for v in value.lstrip("[").rstrip(")").split(", ")]
-            condition = {
-                f"{field}__gte": start,
-                f"{field}__lt": end
-            }
+            if json_range_condition := f_utils.get_json_range_condition(field_props, start, end):
+                # JSONField array range stats must use 'jsonb_path_exists' conditions
+                condition = json_range_condition
+            else:
+                condition = Q(**{
+                    f"{field}__gte": start,
+                    f"{field}__lt": end
+                })
         else:
             [sym, val] = value.split(" ")
             if sym == "≥":
-                condition = {f"{field}__gte": int(val)}
+                if json_range_condition := f_utils.get_json_range_condition(field_props, min=int(val)):
```

Comment:
just checking, this is false-y when `Q()` is empty?

Reply:
Yes, the `__bool__` function of `Q` is `len(children)`, so it is false when it is empty